datafusion/optimizer/src/limit_push_down.rs

```diff
@@ -192,6 +196,24 @@ impl OptimizerRule for LimitPushDown {
             LogicalPlan::Join(join) => {
                 let limit = fetch + skip;
                 let new_join = match join.join_type {
+                    JoinType::Left | JoinType::Right | JoinType::Full
+                        if is_no_join_condition(join) =>
```

Comment:
I wonder how often there will be no join conditions 🤔  that effectively then becomes a CROSS JOIN

Reply:
> I wonder how often there will be no join conditions 🤔 that effectively then becomes a CROSS JOIN
> 
> To be clear I think the optimization is still correct, I just wonder how often it will appear in real queries

Good question for this, in the current framework of optimizer in the datafusion, the `join` query will be converted to the `cross join` by other rule.

The `limit push down` rule just do the best to optimize the plan.

Some SQL are generated by the program, and the `join type` and `on condition` are added by the program, it will generate sql like ` select * from left_table right join right_table`.